### PR TITLE
Winding/Orientation for LineStrings

### PR DESCRIPTION
**Pull request by rory**
After feedback on #132, I've redid this, using the `Orient` trait.

The `Orient` trait is very useful for encoding (Multi)Polygons, since you can ensure the data is correctly ordered, but it's not helpful for reading bytes. In that case, you need a way to look at a linestring and see if it's clockwise or counter-clockwise, to see if it's an interior or exterior ring. That's what I've added in this PR. Since this functionality only makes sense for LineStrings, I implemented it as methods on the LineString struct, rather than create a trait.

Taking advice from #132, I've made `.winding_order()` return `Option<WindingOrder>`.

I also added some shorthand methods to make it easier to call (e.g. `.orient_default()`)

### CODE REVIEW

**Comment by rory** [comment]
Any feedback on this?

File: src/algorithm/orient.rs [hunk 1]
```diff
@@ -29,6 +29,20 @@ pub trait Orient<T> {
     /// assert_eq!(oriented.interiors[0].0, oriented_int_ls.0);
     /// ```
     fn orient(&self, orientation: Direction) -> Self;
+
+    /// Orient a Polygon according to default orientation
+    fn orient_default(&self) -> Self
```
Comment (frewsxcv) [inline src/algorithm/orient.rs]: I personally think it's a little redundant having these new methods in addition to `orient`, but if people find it helpful then I'm cool with it

File: src/algorithm/orient.rs [hunk 2]
```diff
@@ -61,8 +75,14 @@ pub enum Direction {
     Reversed,
 }
 
+#[derive(PartialEq, Copy, Clone, Debug)]
```
Comment (frewsxcv) [inline src/algorithm/orient.rs]: This could also derive `Eq`

File: src/algorithm/orient.rs (hunk 1 above)
Comment (urschrei) [reply src/algorithm/orient.rs]: Yeah, I think it should be one or the other – I'm not tied to using an enum, though.

**Comment by rory** [comment]
OK, those are good points. I'll reimplent this feature this way shortly and update this pull request. Stay tuned!

**Comment by rory** [comment]
continued in #169

**Pull request closed without merging**